library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt

```diff
@@ -0,0 +1,162 @@
+package com.applandeo.materialcalendarview.listeners
+
+import android.view.View
+import android.widget.AdapterView
+import android.widget.AdapterView.OnItemClickListener
+import android.widget.TextView
+import com.applandeo.materialcalendarview.CalendarView
+import com.applandeo.materialcalendarview.EventDay
+import com.applandeo.materialcalendarview.R
+import com.applandeo.materialcalendarview.adapters.CalendarPageAdapter
+import com.applandeo.materialcalendarview.getDatesRange
+import com.applandeo.materialcalendarview.utils.*
+import java.util.*
+
+/**
+ * This class is responsible for handle click events
+ *
+ *
+ * Created by Applandeo Team.
+ */
+class DayRowClickListener(
+        private val calendarPageAdapter: CalendarPageAdapter,
+        private val calendarProperties: CalendarProperties,
+        pageMonth: Int
+) : OnItemClickListener {
+
+    private val pageMonth = if (pageMonth < 0) 11 else pageMonth
+
+    override fun onItemClick(adapterView: AdapterView<*>, view: View, position: Int, id: Long) {
+        val day = GregorianCalendar().apply {
+            time = adapterView.getItemAtPosition(position) as Date
+        }
+
+        if (calendarProperties.selectionDisabled) return
+
+        if (calendarProperties.onDayClickListener != null) {
+            onClick(day)
+        }
+
+        when (calendarProperties.calendarType) {
+            CalendarView.ONE_DAY_PICKER -> selectOneDay(view, day)
+            CalendarView.MANY_DAYS_PICKER -> selectManyDays(view, day)
+            CalendarView.RANGE_PICKER -> selectRange(view, day)
+            CalendarView.CLASSIC -> calendarPageAdapter.selectedDay = SelectedDay(day, view)
+        }
+    }
+
+    private fun selectOneDay(view: View, day: Calendar) {
+        val previousSelectedDay = calendarPageAdapter.selectedDay
+
+        val dayLabel = view.findViewById<TextView>(R.id.dayLabel)
+
+        if (isAnotherDaySelected(previousSelectedDay, day)) {
+            selectDay(dayLabel, day)
+            reverseUnselectedColor(previousSelectedDay)
+        }
+    }
+
+    private fun selectManyDays(view: View, day: Calendar) {
+        val dayLabel = view.findViewById<TextView>(R.id.dayLabel)
+
+        if (day.isCurrentMonthDay() && day.isActiveDay()) {
```

Comment:
you can do it in the same way like below:

`if (!day.isCurrentMonthDay() || !day.isActiveDay()) return`